### PR TITLE
Empty string 2D Array test case and classes

### PR DESCRIPTION
**Pull request by MakarovAlexey**
Pull Request for the #161 issue

### CODE REVIEW

**Comment by vladmihalcea** [comment]
I haven't tested the current array types as multidimensional arrays. Is this Pull Request a prototype for that?

**Comment by MakarovAlexey** [comment]
Yes, Pull Request contains required classes ('*.internal.String2DArrayTypeDescriptor' and 'String2DArrayType').

**Comment by vladmihalcea** [comment]
> Pull Request contains new classes only for two dimensional case

Is this a fully functional example for multidimensional arrays or is it a work in progress which generates the #161 issue?

**Comment by MakarovAlexey** [comment]
This is work in progress which generated the #161 isse.

**Comment by vladmihalcea** [comment]
You will have to debug it and see why it does not work. Once you have a fully-functional solution, I'll review it to see whether it can be integrated. Good luck.

**Comment by MakarovAlexey** [comment]
In 'wrap' method of 'AbstractArrayTypeDescriptor', the method 'getArray' of 'java.sql.Array' return one-dimensional empty array when postgresql array is empty. I found workaround by overriding 'wrap' in 'String2DArrayTypeDescriptor'.
Casting to String[][] directly throws type cast exception.

File: hibernate-types-52/src/main/java/com/vladmihalcea/hibernate/type/array/internal/String2DArrayTypeDescriptor.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    @Override
+    protected String getSqlArrayType() {
+        return "text";
+    }
+
+    @Override
+    public <X> String[][] wrap(X value, WrapperOptions options) {
+        Object[] array = super.wrap(value, options);
```
Comment (MakarovAlexey) [inline hibernate-types-52/src/main/java/com/vladmihalcea/hibernate/type/array/internal/String2DArrayTypeDescriptor.java]: Casting to String[][] directly throws type cast exception.

**Comment by vladmihalcea** [comment]
I'll review it when I have some time. Thanks for your contribution.

**Comment by vladmihalcea** [comment]
No need to create new types. Only the array initialization needed a change. Check out [this commit](https://github.com/vladmihalcea/hibernate-types/commit/b3d77a4fc1c69d0880156a4426ae66294902428a) for more details.

**Pull request closed without merging**